File: entities/idempotency_record.go
package entities

import "time"

type IdempotencyRecord struct {
	Key        string    `json:"key" gorm:"primaryKey;size:255"`
	Response   string    `json:"response" gorm:"type:text;not null"`
	StatusCode int       `json:"statusCode" gorm:"column:status_code;not null"`
	CreatedAt  time.Time `json:"createdAt" gorm:"column:created_at"`
}

File: ts_solution/src/entities/IdempotencyRecord.ts
import {
  Entity,
  PrimaryColumn,
  Column,
  CreateDateColumn,
} from "typeorm";

@Entity("idempotency_records")
export class IdempotencyRecord {
  @PrimaryColumn({ length: 255 })
  key!: string;

  @Column({ type: "text" })
  response!: string;

  @Column({ type: "integer" })
  statusCode!: number;

  @CreateDateColumn({ name: "created_at" })
  createdAt!: Date;
}
